### PR TITLE
Makes plastic flap, airshield and beach water mouse clicking easier to deal with

### LINKED ISSUE #36454: Smarter mouse opacity
image
Let's take this image as an example. These are airshields, which block airflow on a whole tile and are pretty incredible for building projects. Unfortunately, because they are an overlay, they actually make it impossible to easily click on objects below. There are several objects and tiles like this--big fully-enclosing objects which block your mouse despite being transparent. Water tiles are another example.

Well, what we can do is turn their `mouse_opacity` to `0`, which makes it so that your mouse *always* ignores them. This means you can click on the objects below, and the turf below. However, you *can't* click on the airshield itself, nor can you mouse over it to see what it even is.

So this seems like a problem without a perfect solution, right? Either we can't click on anything below the overlay, or we can't click on the overlay at all. Barring right clicks and alt clicks to see what's on the tile, of course.

What if there was a more sophisticated system for mouse opacity? 

For instance, we could make only the outside of the airshield's sprite have mouse opacity, since that's the part that isn't actually *transparent*.

image

We could also dream up an instance where mouse opacity has a priority system. For instance, water tiles overlay over objects to give the impression that the objects are submerged, but that means you can't click on them. If you give the tiles `mouse_opacity 0`, that means you can't even examine the water or click on it. What if your mouse always ignored the tile when aiming at objects or mobs on top of it, but if your mouse goes through everything and only overlaps the tile, you could see that it was water and click on it?

image

That latter method could work on airshields too--being able to click on the airshield when it only has turf below, but allowing you to click on items below.

image

Here's a final example that always gets my goat. I think that plastic flaps would be a prime candidate to have `mouse_opacity 0` on certain parts of the sprite, so that you can properly trade and put down items in circumstances where they're placed on top of tables. If only the exterior had mouse opacity, it would make moving them around or deconstructing them a little harder, but it would make general use (including crawling under them) really easy.

Thanks for coming to my girthhub talk

[featurerequest] [qol]

### PR DESCRIPTION
**Pull request by SECBATON-GRIFFON**
[qol]

## What this does
makes the inner two strips on plastic flaps not mouse opaque, as a BYOND level compromise to make it possible to put stuff on tables below it easier but also easy to deconstruct from the sides.
does this for airshields too, allows only the machinery part edges to be interacted with now.
does this for beach water entirely since it was just an overlay on a turf anyways.
all achieved with vis_contentsing in another object with no mouse opacity, only makes one global one for them as to avoid reuse.
Closes #36454.

## Why it's good
interacting with underneaths is easier now.

## Changelog
:cl:
 * tweak: It's now easier to interact with items under plastic flaps, airshields or beach water, with the compromise of any interactions with those items themselves being pushed to the sides of them with the exception of beach water.

### CODE REVIEW

**Comment by west3436** [comment]
Edit

**Comment by west3436** [comment]
Disregard my previous comments - this does work.

File: code/game/objects/structures/flaps.dm [hunk 1]
```diff
@@ -9,6 +9,27 @@
 	pass_flags_self = PASSGLASS
 	var/airtight = 0
 
+/obj/effect/plasticflaps_overlay
+	icon = 'icons/obj/stationobjs.dmi' //Change this.
```
Comment (west3436) [inline code/game/objects/structures/flaps.dm:13]: //Change this.
???

Comment (SECBATON-GRIFFON) [reply code/game/objects/structures/flaps.dm:13]: just matching the comment above on line 4

**Pull request merged**